lib/format_parser.rb

```diff
@@ -30,24 +39,30 @@ def self.parse_http(url)
     parse(cached_io)
   end
 
-  def self.parse(io)
+  def self.parse(io, **opts, &proc)
+    config = parse_config(**opts, &proc)
     # If the cache is preconfigured do not apply an extra layer. It is going
     # to be preconfigured when using parse_http.
     io = Care::IOWrapper.new(io) unless io.is_a?(Care::IOWrapper)
+    results = []
 
     # Always instantiate parsers fresh for each input, since they might
     # contain instance variables which otherwise would have to be reset
     # between invocations, and would complicate threading situations
-    parsers = @parsers.map(&:new)
+    parsers = @parsers.select do |p|
+      !(p.natures & config.natures).empty? && !(p.formats & config.formats).empty?
+    end.map(&:new)
```

Comment:
Do we really need to fence off the question methods about what parsers support in their class method namespace? Isn't it simpler to query parsers after instantiating them? We will create throwaway instances while doing so, but this will prevent too much bookkeeping within the parsers - of which we have many.

Reply:
Also there's quite a bit happening here - if we _do_ need something like this I think it could be made more explicit. (Maybe just my personal styling pref though.)